src/client/pages/OfferNew/index.tsx

```diff
@@ -59,18 +89,54 @@ export const OfferNew: React.FC = () => {
     skip: quoteIdsIsLoading,
   })
 
+  const bundleVariants = React.useMemo(
+    () =>
+      (data?.quoteBundle.possibleVariations as Array<QuoteBundleVariant>) ?? [],
+    [data?.quoteBundle.possibleVariations],
+  )
+  const canShowInsuranceSelector =
+    isInsuranceToggleEnabled && bundleVariants.length > 1
+  const [selectedBundleVariant, setSelectedBundleVariant] = React.useState<
+    QuoteBundleVariant
+  >()
+
+  useEffect(() => {
+    if (!selectedBundleVariant) {
+      return setSelectedBundleVariant(
+        getBundleVariantFromQuoteIds(selectedQuoteIds, bundleVariants),
+      )
+    }
+
+    const matchingVariant = bundleVariants.find(
+      ({ id }) => id === selectedBundleVariant.id,
+    )
+    if (!matchingVariant) return setSelectedBundleVariant(bundleVariants[0])
+    return setSelectedBundleVariant(matchingVariant)
```

Comment:
I'm starting to think we should rethink here. It's better to derive (useMemo) rather than syncing (useEffect).

Can't we just derive the `selectedBundleVariant` from the `selectedQouteIds`?